code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+     *
+     * @param event the event requesting/triggering an update to configuration with appId.
+     */
+    private fun configureWithAppID(event: Event) {
+        val appID = event.eventData[CONFIGURATION_REQUEST_CONTENT_JSON_APP_ID] as? String
+
+        if (appID.isNullOrBlank()) {
+            appIdManager.removeAppIDFromPersistence()
+
+            publishConfigurationState(
+                configurationStateManager.environmentAwareConfiguration,
+                event
+            )
+            return
+        }
+
+        if (!configurationStateManager.hasConfigExpired(appID)) {
+            publishConfigurationState(
+                configurationStateManager.environmentAwareConfiguration,
+                event
+            )
+            return
+        }
+
+        // Stop all event processing for the extension until new configuration is downloaded
+        api.stopEvents()
+
+        configurationStateManager.updateConfigWithAppId(appID) { config ->
+            if (config != null) {
+                publishConfigurationChanges(event, false)
+                // re-start event processign after new configuration download
+                api.startEvents()
+            } else {
+                // If the configuration download fails, retry download again.
+                retryWorker.schedule({ configureWithAppID(event) }, CONFIG_DOWNLOAD_RETRY_ATTEMPT_DELAY_MS, TimeUnit.SECONDS)
+            }
+        }
+    }
+
+    /**
+     * Retrieves the configuration from the file path specified in the [event]'s data and replaces
+     * the current configuration with it.
+     *
+     * @param event the trigger event (whose event data contains the file path for retrieving the config)
+     *              requesting a configuration change
+     */
+    private fun configureWithFilePath(event: Event) {
+        val filePath = event.eventData[CONFIGURATION_REQUEST_CONTENT_JSON_FILE_PATH] as String?
+
+        if (filePath.isNullOrBlank()) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Unable to read config from provided file (filePath is invalid)"
+            )
+            return
+        }
+
+        val config = configurationStateManager.getConfigFromFile(filePath)
+        if (config == null) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Unable to read config from provided file (content is invalid)"
+            )
+            return
+        }
+
+        configurationStateManager.replaceConfiguration(config)
+        publishConfigurationChanges(event, false)
+    }
+
+    /**
+     * Updates the current configuration with the content from [ConfigurationStateManager.CONFIG_BUNDLED_FILE_NAME]
+     *
+     * @param event which contains [CONFIGURATION_REQUEST_CONTENT_JSON_ASSET_FILE] as part of its event data
+     */
+    private fun configureWithFileAsset(event: Event) {
```

Comment:
Isn't there a public API which also accepts the `fileName`? 

https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/configuration/configuration-api-reference#configurewithfileinassets 

https://git.corp.adobe.com/dms-mobile/bourbon-core-java-core/blob/e6f4e153ef00cb053f699d453e97272ab0db8ca5/code/configuration/src/main/java/com/adobe/marketing/mobile/ConfigurationExtension.java#L312

Reply:
Is fetching the `fileName` from the  `event` missing? Looks like we always use `CONFIG_BUNDLED_FILE_NAME`

Reply:
You are right! I missed the two public api's for configuration as well as fetching the asset name from the event. Will change accordingly.